### PR TITLE
Return http response error

### LINKED ISSUE #21: Bette error messages
The current error handling simply prints the raw response of the HTTP response. For example, if you do
```go
db.Query("SELECT 1") // without ";" in the end
```
The raw response will look like this
```
rsp: &{400 Bad Request 400 HTTP/1.1 1 1 map[Date:[Fri, 24 May 2019 05:39:06 GMT] Content-Type:[application/xml] Content-Length:[227] Content-Md5:[53a2c43ea4f4738421503fae389b762a] X-Odp
s-Request-Account:[1829387660937356] X-Odps-Request-Id:[5CE7837983C0893F7E086331] Server:[Tengine/2.1.16_20180517093604]] 0xc00044c880 227 [] true false map[] 0xc00028a200 0xc0004ca0b0} 
```
which is not very helpful for debugging the query.

### LINKED ISSUE #34: More readable error message on parsing http response
The error message following could be more readable
```
    connection_test.go:17: 
                Error Trace:    connection_test.go:17
                Error:          Received unexpected error:
                                response error: 400. json error: invalid character '<' looking for beginning of value
                                sqlflow.org/gomaxcompute.parseResponseBody
                                        /Users/weiguo/go/src/sqlflow.org/gomaxcompute/client.go:173
                                sqlflow.org/gomaxcompute.(*odpsConn).createInstance
                                        /Users/weiguo/go/src/sqlflow.org/gomaxcompute/instance.go:30
                                sqlflow.org/gomaxcompute.(*odpsConn).wait
                                        /Users/weiguo/go/src/sqlflow.org/gomaxcompute/connection.go:128
                                sqlflow.org/gomaxcompute.(*odpsConn).Query
                                        /Users/weiguo/go/src/sqlflow.org/gomaxcompute/connection.go:58
                                database/sql.ctxDriverQuery
                                        /usr/local/Cellar/go/1.12.6/libexec/src/database/sql/ctxutil.go:60

```

### PR DESCRIPTION
**Pull request by weiguoz**
fix #34 
fix #21 

After this PR, 400 error:
```
response error: 400, <?xml version="1.0" encoding="UTF-8"?>
                                <Error>
                                        <Code>InvalidParameter</Code>
                                        <Message><![CDATA[Query must ends with ';']]></Message>
                                        <RequestId>5D04527654CF64D877B33F89</RequestId>
                                        <HostId>odps.aliyun.com</HostId>
                                </Error>
```

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -167,12 +168,21 @@ func parseResponseBody(rsp *http.Response) ([]byte, error) {
 	defer rsp.Body.Close()
 	body, err := ioutil.ReadAll(rsp.Body)
 
-	if rsp.StatusCode >= 400 {
-		re := responseError{}
-		if err = json.Unmarshal(body, &re); err != nil {
-			return nil, errors.WithStack(fmt.Errorf("response error: %d. json error: %v", rsp.StatusCode, err))
-		}
-		return nil, errors.WithStack(fmt.Errorf("response error: %d, %s. %s", rsp.StatusCode, re.Code, re.Message))
+	if err == nil {
```
Comment (tonyyang-svail) [inline client.go]: `if err == nil` is a rare find. How about
```go
if err != nil {
	return nil, error
}

if rsp.StatusCode >= 400 {
	return nil, parseResponseError(...)
}

return body, err
```

File: client.go [hunk 2]
```diff
@@ -166,13 +167,21 @@ func parseResponseBody(rsp *http.Response) ([]byte, error) {
...
-		}
-		return nil, errors.WithStack(fmt.Errorf("response error: %d, %s. %s", rsp.StatusCode, re.Code, re.Message))
+		return nil, maxcomputeResponseError(rsp.StatusCode, body)
 	}
 	return body, err
 }
+
+func maxcomputeResponseError(statusCode int, body []byte) error {
+	re := responseError{}
+	if err := json.Unmarshal(body, &re); err == nil {
```
Comment (tonyyang-svail) [inline client.go]: Similarly, `err != nil`.

File: client.go [hunk 3]
```diff
@@ -166,13 +167,21 @@ func parseResponseBody(rsp *http.Response) ([]byte, error) {
...
-		if err = json.Unmarshal(body, &re); err != nil {
-			return nil, errors.WithStack(fmt.Errorf("response error: %d. json error: %v", rsp.StatusCode, err))
-		}
-		return nil, errors.WithStack(fmt.Errorf("response error: %d, %s. %s", rsp.StatusCode, re.Code, re.Message))
+		return nil, maxcomputeResponseError(rsp.StatusCode, body)
 	}
 	return body, err
 }
+
+func maxcomputeResponseError(statusCode int, body []byte) error {
```
Comment (tonyyang-svail) [inline client.go]: I feel `maxcompute` is redundant...

File: client.go [hunk 4]
```diff
@@ -163,16 +164,25 @@ func parseResponseBody(rsp *http.Response) ([]byte, error) {
 	if rsp == nil || rsp.Body == nil {
 		return nil, errNilBody
 	}
+  log.Debugf("response code: %v", rsp.StatusCode)
```
Comment (tonyyang-svail) [inline client.go]: indentation?

File: client.go (hunk 3 above)
Comment (weiguoz) [reply client.go]: `responseError` is a `struct` name here already

Comment (tonyyang-svail) [reply client.go]: How about `parseResponseError`?

Comment (weiguoz) [reply client.go]: Agree

**Review by tonyyang-svail — APPROVED** [review]
LGTM!

**Pull request merged**